asv/plugins/virtualenv.py

```diff
@@ -124,13 +124,13 @@ def install_requirements(self):
 
         self.setup()
 
-        self.upgrade('setuptools==3.8')
+        self._upgrade('setuptools==3.8')
 
         for key, val in six.iteritems(self._requirements):
             if val is not None:
-                self.upgrade("{0}=={1}".format(key, val))
+                self._upgrade("{0}=={1}".format(key, val))
             else:
-                self.upgrade(key)
+                self._upgrade(key)
```

Comment:
I think we can do the same thing for `virtualenv` that was done for `conda`, i.e. pass all dependencies in one go.  Then we'll have fixed the original bug in both places.